source/agora/utils/Test.d

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+
+    Various utilities for testing purpose
+
+    Utilities in this module can be used in test code.
+    There are currently multiple testing approaches:
+    - Unittests in the various `agora` module, the most common, cheapest,
+      and a way to do white box testing;
+    - Unittests under `agora.test`: Those unittests rely on the LocalRest
+      library to simulate a network where nodes are thread who communicate
+      via message passing.
+    - Unit integration tests in `${ROOT}/tests/unit/` which are similar to
+      unittests but provide a way to test IO-using code.
+    - System integration tests: those are fully fledged tests that spawns
+      unmodified, real nodes within Docker containers and act as a client.
+
+    Any symbol in this module can be used by any of those method,
+    which is why this module is neither restricted by `package(agora):`
+    nor `version(unittest):`.
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.utils.Test;
+
+import std.file;
+import std.path;
+
+/*******************************************************************************
+
+    Get a temporary directory for unit integration tests
+
+    Tests that do IO usually write or read files from disk.
+    We want our tests to be reliable, reproducible, and re-runnable.
+    For this reason, this function returns a path which has been `mkdir`ed
+    after having been cleaned, which is located in the temporary directory.
+    Consistent usage of this allows unit integration tests to be run in parallel
+    (however the same test cannot be run multiple times in parallel,
+    unless a different postfix is specified each time).
+
+    Params:
+      postfix = A unique postfix for the calling test
+
+    Returns:
+      The path of a clean, empty directory
+
+*******************************************************************************/
+
+public string makeCleanTempDir (string postfix = __MODULE__)
```

Comment:
Probably the most documented function in agora.

@bpfkorea/core : There is a neat trick here. Default parameters are evaluated at the call site. So when this function is called without argument, the module will be `"unit.BlockStorage"`, `"unit.BanManager"`, etc...

If the default parameter was evaluated in this module, the result would always be `"agora.utils.Test"`...
This trick is used in many places, most commonly with `__LINE__`, `__FILE__`, and of course `__MODULE__`.